pyfar/classes/audio.py

```diff
@@ -1021,7 +1039,122 @@ def power(data: tuple, domain='freq'):
     return _arithmetic(data, domain, _power)
 
 
-def _arithmetic(data: tuple, domain: str, operation: Callable):
+def matrix_multiplication(
+        data: tuple, domain='freq', axes=[(-2, -1), (-2, -1), (-2, -1)]):
+    """Matrix multiplication of multidimensional pyfar audio objects and/or
+    array likes.
+
+    This function is based on ``numpy.matmul`` and only acts on the channels,
+    not on the frequency/time dimension. Pyfar audio objects are:
+    :py:func:`Signal`, :py:func:`TimeData`, and :py:func:`FrequencyData`.
+
+    Parameters
+    ----------
+    data : tuple of the form (data_1, data_2, ..., data_N)
+        Data to be multiplied. Can contain pyfar audio objects or array likes.
+        The data is `broadcasted
+        <https://numpy.org/doc/stable/user/basics.broadcasting.html>`_
+        to the data with the highest number of channel dimensions. Pyfar audio
+        objects can not be mixed, e.g., :py:func:`TimeData` and
+        :py:func:`FrequencyData` objects do not work together. See
+        :py:mod:`arithmetic operations <pyfar._concepts.arithmetic_operations>`
+        for possible combinations of Signal FFT normalizations.
+    domain : ``'time'``, ``'freq'``, optional
+        Flag to indicate if the operation should be performed in the time or
+        frequency domain. If working in the frequency domain, the FFT
+        normalization is removed before the operation (See
+        :py:func:`~pyfar.dsp.fft.normalization`). The default is ``'freq'``.
+    axes : list of 3 tuples
+        A list of tuples with the indices of the channels the
+        multiplication should operate on. The signature is (i,j),(j,k)->(i,k),
+        so the base elements of the multiplication are two-dimensional
+        matrices along the time/frequency dimension taken to be stored in the
+        indicated channels of the result. Negative values in ``axes`` refer to
+        ``cshape``, as the frequency/time axis is excluded. For instance, to
+        multiply matrices residing (stacked) in the last two channel
+        dimensions to be stored in the last two channel dimensions, this is
+        indicated by ``[(-2, -1), (-2, -1), (-2, -1)]``, which is the default.
+        See notes and examples below.
```

Comment:
This was not clear upon first reading to me. Should we make more clear that the first tuple gives the axes (defines the matrix) for the first input, the second axes do the same for the second input and the third define in what axes the result is saved?